octreelib/grid/grid.py

```diff
@@ -107,23 +108,118 @@ def insert_points(self, pose_number: int, points: PointCloud):
             self.__pose_voxel_coordinates[pose_number].append(target_voxel)
             self.__octrees[target_voxel].insert_points(pose_number, voxel_points)
 
-    def map_leaf_points(self, function: Callable[[PointCloud], PointCloud]):
+    def map_leaf_points(
+        self,
+        function: Callable[[PointCloud], PointCloud],
+        pose_numbers: Optional[List[int]] = None,
+    ):
         """
         Transform point cloud in each node of each octree using the function
         :param function: Transformation function PointCloud -> PointCloud. It is applied to each leaf node.
+        :param pose_numbers: List of pose numbers to map.
         """
         for voxel_coordinates in self.__octrees:
-            self.__octrees[voxel_coordinates].map_leaf_points(function)
+            self.__octrees[voxel_coordinates].map_leaf_points(function, pose_numbers)
 
-    def get_leaf_points(self, pose_number: int) -> List[Voxel]:
+    def map_leaf_points_cuda_ransac(
+        self,
+        n_poses_per_batch: int = 1,
+        threshold: float = 0.01,
+        n_hypotheses: int = 1024,
+    ):
+        """
+        transform point cloud in the node using the function
+        :param n_poses_per_batch: Number of poses per batch.
+        :param threshold: Distance threshold.
+        :param n_hypotheses: Number of RANSAC iterations (<= 1024).
```

Comment:
i guess it would be clear if you add assert on this case

Reply:
i've added a check with an error message in case of bad values for these parameters.
can change to an assert if necessary

Reply:
Yeah, I forgot that there are Exceptions exist :smile: